test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -518,61 +519,57 @@ export default class TabularView extends Component {
         }
     }
 
-    populateCompTable(parserProps) {
+    populateCompTable = async() => {
     	// Each entry is a combination of the data in testData and baselineData, same format two fields benchmarkName and platforms
     	// Now platform entries contain information from both the testJdk and the baselineJdk
         const newArray = [];
-        var metricProps;
-        this.state.testData.forEach(function (testDataObject) {
+        for(let testDataObject of this.state.testData) {
             let consolidatedDataObject = {};
             let benchmark = testDataObject.benchmarkNVM.split(",")[0];
             let metric = testDataObject.benchmarkNVM.split(",")[2];
             consolidatedDataObject.platformsSpecificData = {};
             consolidatedDataObject.benchmarkNVM = testDataObject.benchmarkNVM;
-
-            let higherBetter;
-            
-            //get BenchmarkRouter & Metric files from server
-            metricProps = getMetricProps(parserProps, benchmark, metric);
-            try {
-                if (metricProps["higherbetter"]) {
-                    higherBetter = true;
-                } else {
-                    higherBetter = false;
+            //To get the values of highterbetter/units  
+            //first check if Metric does already exist in constructor , if not get its info from server
+            let metricProps;
+            if ( !this.metricsProps[benchmark] ) {
+                const metricPropsJSON = await getBenchmarkMetricProps(benchmark);
+                if(metricPropsJSON){
+                    this.metricsProps[benchmark] = metricPropsJSON;
+                    metricProps = this.metricsProps[benchmark][metric];
                 }
-            } catch (higherBetterNotFoundError) {
-                higherBetter = true;
+            } else {
+                metricProps = this.metricsProps[benchmark][metric];
             }
-
-
+            const higherBetter = (metricProps && (metricProps.higherbetter != null)) ?  metricProps.higherbetter : true;
```

Comment:
There is no need to specifically check `metricProps.higherbetter != null`. Please update the above line according to https://github.com/AdoptOpenJDK/openjdk-test-tools/pull/214#discussion_r400308520

Reply:
`!metricProps || metricProps.higherbetter !== false`